e2e_tests/tests/streaming/test_client.py

```diff
@@ -0,0 +1,76 @@
+import uuid
+
+import pytest
+
+from determined.common.api import bindings
+from determined.common.streams import wire
+from determined.common.streams._client import LomondStreamWebSocket, ProjectSpec, Stream, Sync
+from tests import api_utils
+
+
+@pytest.mark.e2e_cpu_streaming
+def test_client_connection() -> None:
+    sess = api_utils.admin_session()
+    ws = LomondStreamWebSocket(sess)
+    stream = Stream(ws)
+    syncId = "sync1"
+    stream.subscribe(sync_id=syncId, projects=ProjectSpec(workspace_id=1))
+    event = next(stream)
+    assert event == Sync(syncId, False)
+
+    event = next(stream)
+    assert event.id == 1
+    assert event.immutable == True
+    event = next(stream)
+    assert event == Sync(syncId, True)
+
+
+@pytest.mark.e2e_cpu_streaming
+def test_client_connection() -> None:
+    sess = api_utils.admin_session()
+    ws = LomondStreamWebSocket(sess)
+    stream = Stream(ws)
+
+    syncId = "sync1"
+    projectName = "streaming_project"
+    newProjectName = "streaming_project_1"
+
+    resp_w = bindings.post_PostWorkspace(
+        sess, body=bindings.v1PostWorkspaceRequest(name=f"workspace_aug_{uuid.uuid4().hex[:8]}")
+    )
+    w = resp_w.workspace
+    resp_p = bindings.post_PostProject(
+        sess,
+        body=bindings.v1PostProjectRequest(
+            name=projectName,
+            workspaceId=w.id,
+        ),
+        workspaceId=w.id,
+    )
+    p = resp_p.project
+
+    stream.subscribe(sync_id=syncId, projects=ProjectSpec(workspace_id=w.id))
+    event = next(stream)
+    assert event == Sync(syncId, False)
+    event = next(stream)
+    assert event.id == p.id
+    assert event.name == projectName
+    seq = event.seq
+    event = next(stream)
+    assert event == Sync(syncId, True)
+
+    bindings.patch_PatchProject(sess, body=bindings.v1PatchProject(name=newProjectName), id=p.id)
+    event = next(stream)
+    assert event.id == p.id
+    assert event.name == newProjectName
+    assert event.seq > seq
+
+    bindings.delete_DeleteProject(sess, id=p.id)
+    for event in stream:
+        if isinstance(event, wire.ProjectMsg):
+            assert event.state == "DELETING"
+        elif isinstance(event, wire.ProjectsDeleted):
+            assert event == wire.ProjectsDeleted(str(p.id))
+            break
+        else:
+            raise Exception("Unexpected message from stream.")
```

Comment:
don't raise a bare Exception.  Use a `ValueError` if you have to do something whose meaning doesn't convey much.

Also, please include the message in the exception, so that if anybody were to hit the error in CI they'd see the message that caused the problem in its entirety.